sls-versions/src/main/java/com/palantir/sls/versions/SlsVersionMatcher.java

```diff
@@ -165,11 +168,8 @@ public int compare(OrderableSlsVersion version) {
     }
 
     private static OptionalInt parseInt(String maybeInt) {
-        try {
-            return OptionalInt.of(Integer.parseInt(maybeInt));
-        } catch (NumberFormatException e) {
-            return OptionalInt.empty();
-        }
+        Integer maybeInteger = Ints.tryParse(maybeInt);
+        return maybeInteger != null ? OptionalInt.of(maybeInteger) : OptionalInt.empty();
```

Comment:
double-boxing here is unfortunate, but it's a drop in the bucket.
`(int) 1` -> `Integer(1)` -> `OptionalInt(1)`

Reply:
Yep, not a lot we can do without reimplementing that ourselves.